### ISSUE TITLE
Poolboy should notify workers when they are returned to the pool

### ISSUE THREAD

**Issue by asabil** [devinus/poolboy#76]
When working with a database such as postgresql, the workers owning the connection need to be notified when returned to the pool in order to give the worker the opportunity to reset the state of the connection (rollback pending transactions for example).

**Comment by bullno1** [comment]
:+1:

**Comment by juhlig** [comment]
Yeah, this is just what I'm wishing for right now. Maybe an (optional) callback in `poolboy_worker` would be the best thing to do, which would be called on `checkin`. The same could be done for `checkout`, though this would probably be of little use.

**Comment by juhlig** [comment]
Or a function to call that could be given as a pool arg.
